### PR TITLE
Allow cross origin when requesting to the hornet API

### LINKED ISSUE #386: CORS header ‘Access-Control-Allow-Origin’ missing
When using the browser to do API requests to the Hornet [Core](https://github.com/iotaledger/tips/blob/main/tips/TIP-0025/tip-0025.md) and [Indexer](https://github.com/iotaledger/tips/blob/main/tips/TIP-0026/tip-0026.md)  REST API to nodes installed by this installer, the API request is denied.
The reason given by the browser is  ‘Reason: CORS header ‘Access-Control-Allow-Origin’ missing’. 
One can see an example of this [here](https://eddytheco.github.io/NftMinter/).
In that case, one needs to set the Access-Control-Allow-Origin header’s value as explained [here](https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS/Errors/CORSMissingAllowOrigin).

In my opinion, the nodes installed by this installer should respond to browser requests by default.
For this one needs to add
```
- "traefik.http.middlewares.cors.headers.customResponseHeaders.Access-Control-Allow-Origin=*"
- "traefik.http.routers.hornet.middlewares=cors"
```
to `docker-compose.yml` of the Hornet nodes.

Then every node maintainer is free to deny API requests from the browser, maybe by configuration variable of the installer or just deleting the lines we added. 

I could do a PR with the need changes if you consider this is need it.
Best,
Eduardo.

### PR DESCRIPTION
**Pull request by EddyTheCo**
Setting 
```
Access-Control-Allow-Origin=https://${HORNET_HOST:?WASP_HOST is missing}:${HORNET_HTTPS_PORT:-443}"
```

Only allows browser requests to the node originating from the same wasp and Hornet origin.
This restricts browser Dapplications from making requests to the Core  API of the node.
By [default](https://github.com/iotaledger/hornet/issues/1900#issuecomment-1671787597) Hornet allows it.
And I do not think you gain much security by restricting the origins when interacting with the node API.

This should fix #386.

### CODE REVIEW

**Review by dlt-green — CHANGES_REQUESTED** [review]
Do we need this change also with wasp or other inx plugins?

**Review by dlt-green — APPROVED** [review]
Thanks for explanation and participation

**Comment by bogenlos** [comment]
I wonder why you removed all headers. Actually it should be sufficient to just set Access-Control-Allow-Origin: *

**Review by dlt-green — CHANGES_REQUESTED** [review]
see above

**Review by bogenlos — APPROVED** [review]

**Comment by dlt-green** [comment]
> I wonder why you removed all headers. Actually it should be sufficient to just set Access-Control-Allow-Origin: *

ℹ️ The desired functionality with maximum security is required; simply removing all headers is not desired.

Waiting for answer, Thanks

**Review by bogenlos — CHANGES_REQUESTED** [review]
Yes, please just set Access-Control-Allow-Origin: * and let's see how it works out.

**Comment by dlt-green** [comment]
Please only adjust what is necessary, who knows whether there won't be one or another plugin in the future. Thanks

**Comment by EddyTheCo** [comment]
To make API requests from the browser works using your settings could leave 
```
- "traefik.http.middlewares.security-headers.headers.frameDeny=true"
- "traefik.http.middlewares.security-headers.headers.contentTypeNosniff=true"
```
If only set Access-Control-Allow-Origin: * it does not work.

**Comment by dlt-green** [comment]
please push an alternative suggestion to get a review with minimal changes that it is running for you. thanks

**Comment by EddyTheCo** [comment]
[addvaryheader] (https://doc.traefik.io/traefik/middlewares/http/headers/#addvaryheader) is not used because `accessControlAllowOriginList` is not set.

[frameDeny](https://doc.traefik.io/traefik/middlewares/http/headers/#framedeny) 
Set frameDeny to true to add the [X-Frame-Options](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-Frame-Options) header with the value of DENY.
But this is to prevent 'that their content is not embedded into other sites' there is nothing to embed on the REST API.

**Comment by dlt-green** [comment]
please push the adapted code ;-)

**Review by 0xKtota — APPROVED** [review]

**Review by jeizinep — APPROVED** [review]

**Review by 0xKtota — APPROVED** [review]

**Review by 2g4y1 — APPROVED** [review]
Looks good for me.

**Review by bogenlos — APPROVED** [review]

**Comment by dlt-green** [comment]
✅external PR accepted (3 approvals necessary | PR getting 5 approvals / no changes requested)
ℹ️will be integrated into the next version after successful testing (dev-branch)
💚thank you for supporting our installer, don't forget to give us a ⭐ if you like it

**Pull request merged**
